testsuite/Main.hs

```diff
@@ -293,31 +338,54 @@ runClashTest = defaultMain $ clashTestRoot
         , runTest "ReduceZero" def
         , runTest "ReduceOne" def
         , runTest "ExtendingNumZero" def
-        , runTest "GenericBitPack" def{clashFlags=["-fconstraint-solver-iterations=15"]}
         , runTest "AppendZero" def
+        , runTest "GenericBitPack" def{clashFlags=["-fconstraint-solver-iterations=15"]}
         , runTest "UnpackUndefined" def{hdlSim=False}
         ]
+#else
+#if __GLASGOW_HASKELL__ >= 865
+      , clashTestGroup "BitVector"
+        [ runTest "Box" def
+        , runTest "BoxGrow" def
+        , runTest "CLZ" def
+        , runTest "RePack" def{hdlSim=False}
+        , runTest "ReduceZero" def
+        , runTest "ReduceOne" def
+        , runTest "ExtendingNumZero" def
+        , runTest "AppendZero" def
+        ]
```

Comment:
Most of these test are also defined above.
This looks like the start of a maintenance nightmare to me.
I think it would be easier to maintain when every testgroup and every test are only at one place in the file.
With some CPP `#if`s around them where required.

```haskell
      , clashTestGroup "BitVector"
        [ runTest "test1" def
        , runTest "test2" def

#if some_condition
        , runTest "test3" def
#endif

#if some_other_condition
        , runTest "test4" def
#endif
        ...
        ]
```

Reply:
Yeah, I'm not so happy with how much chopping and changing has happened in this file. It's also harder to exclude just some tests in a block because of the brackets / commas starting on the beginning of lines :cry: 

I'll try and improve it a bit now (it's gotten worse since I've rebased twice onto master since making this branch) and reduce the number of CPP directives. Fortunately most of these are due to the primitives for things like Natural and Enum not being implemented, so a lot of them can be removed quickly one the primitive evaluator work starts.